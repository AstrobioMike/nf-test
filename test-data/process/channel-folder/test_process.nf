process TEST_PROCESS {

  publishDir "${params.outdir}", mode: 'copy'

  input:
    val number

  output:
     path "folder", emit: my_output_folder

  """
  mkdir folder
  echo "lukas forer" > "folder/${number}_a.txt"
  echo "lukas forer" > "folder/${number}_b.txt"
  echo "lukas forer" > "folder/${number}_c.txt"
  """

}
