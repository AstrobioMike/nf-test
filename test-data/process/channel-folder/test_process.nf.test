nextflow_process {

  name "Test process xy"

  script "test-data/process/channel-folder/test_process.nf"
  process "TEST_PROCESS"

  test("Should create 5 files") {

    when {
      params {
        outdir = "$outputDir"
      }
      process {
        """
        input[0] = Channel.of(1)
        """
      }
    }

    then {
      assert snapshot(process.out).match()
    }

  }

}
