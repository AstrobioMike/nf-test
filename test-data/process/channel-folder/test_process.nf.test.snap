{
    "Should create 5 files": {
        "content": [
            {
                "my_output_folder": [
                    [
                        "1_a.txt:md5,086e29e2872195015ce2d3104909b20f",
                        "1_b.txt:md5,086e29e2872195015ce2d3104909b20f",
                        "1_c.txt:md5,086e29e2872195015ce2d3104909b20f"
                    ]
                ]
            }
        ],
        "timestamp": "2022-10-24T11:17:51+0000"
    }
}